### ISSUE TITLE
handle_vehicle task crashed with an exception on 0.4.0-rc

### ISSUE THREAD

**Issue by Ivesvdf** [SAIC-iSmart-API/saic-python-mqtt-gateway#65]
On [f4d2e27](https://github.com/SAIC-iSmart-API/saic-python-mqtt-gateway/commit/f4d2e27c5111e4e045f6dbf52bdb465ee57970c3) when starting the service it crashes every time with following notice

```
saic-mqtt-gateway  | 2023-08-01 10:08:25,022 Connected to MQTT broker
saic-mqtt-gateway  | 2023-08-01 10:08:25,796 Registering for abnormal messages
saic-mqtt-gateway  | 2023-08-01 10:08:26,099 Registering for moving messages
saic-mqtt-gateway  | 2023-08-01 10:08:26,393 Registering for region messages
saic-mqtt-gateway  | 2023-08-01 10:08:26,684 Registering for engineStart messages
saic-mqtt-gateway  | 2023-08-01 10:08:26,968 Registering for startVehicleStatus messages
saic-mqtt-gateway  | 2023-08-01 10:08:27,263 Registering for offCar messages
saic-mqtt-gateway  | 2023-08-01 10:08:27,558 Registering for speeding messages
saic-mqtt-gateway  | 2023-08-01 10:08:37,612 Setting active query interval in vehicle handler for VIN #VIN# to 0:00:30
saic-mqtt-gateway  | 2023-08-01 10:08:37,614 Setting charging query interval in vehicle handler for VIN #VIN# to 0:00:00
saic-mqtt-gateway  | 2023-08-01 10:08:37,616 Setting after shutdown query interval in vehicle handler for VIN #VIN# to 0:02:00
saic-mqtt-gateway  | 2023-08-01 10:08:37,617 Setting inactive query interval in vehicle handler for VIN #VIN# to 1 day, 0:00:00
saic-mqtt-gateway  | 2023-08-01 10:08:37,619 Setting charging query interval in vehicle handler for VIN #VIN# to 0:00:00
saic-mqtt-gateway  | 2023-08-01 10:08:37,622 Setting refresh mode to periodic
saic-mqtt-gateway  | 2023-08-01 10:08:37,626 Updating vehicle status
saic-mqtt-gateway  | 2023-08-01 10:08:53,812 Updating charging status
saic-mqtt-gateway  | 2023-08-01 10:09:09,945 Removing scheduled check for charging start for VIN #VIN#
saic-mqtt-gateway  | 2023-08-01 10:09:09,968 'handle_vehicle_#VIN#' task crashed with an exception
saic-mqtt-gateway  | Traceback (most recent call last):
saic-mqtt-gateway  |   File "/usr/src/app/./mqtt_gateway.py", line 100, in handle_vehicle
saic-mqtt-gateway  |     charge_status = self.update_charge_status()
saic-mqtt-gateway  |   File "/usr/src/app/./mqtt_gateway.py", line 129, in update_charge_status
saic-mqtt-gateway  |     self.vehicle_state.handle_charge_status(charge_mgmt_data)
saic-mqtt-gateway  |   File "/usr/src/app/vehicle.py", line 513, in handle_charge_status
saic-mqtt-gateway  |     self.update_scheduled_charging(start_time, mode)
saic-mqtt-gateway  |   File "/usr/src/app/vehicle.py", line 163, in update_scheduled_charging
saic-mqtt-gateway  |     self.__scheduler.remove_job(job_id)
saic-mqtt-gateway  |   File "/usr/local/lib/python3.10/site-packages/apscheduler/schedulers/base.py", line 625, in remove_job
saic-mqtt-gateway  |     raise JobLookupError(job_id)
saic-mqtt-gateway  | apscheduler.jobstores.base.JobLookupError: 'No job by the id of #VIN#_scheduled_charging was found'
```

**Comment by nanomad** [comment]
Uhmm, this is a nasty one. I'll have a pr ready later today to fix it

**Comment by Ivesvdf** [comment]
Confirmed fixed in [feddcec](https://github.com/SAIC-iSmart-API/saic-python-mqtt-gateway/commit/feddcec9658d260acbbfa94f1039ea642bccea7e) 


```
saic-mqtt-gateway  | 2023-08-01 18:10:11,660 [  INFO  ] Connected to MQTT broker - mqtt_publisher
saic-mqtt-gateway  | 2023-08-01 18:10:11,665 [  INFO  ] Scheduler started - apscheduler.scheduler
saic-mqtt-gateway  | 2023-08-01 18:10:12,518 [  INFO  ] Registering for abnormal messages - __main__
saic-mqtt-gateway  | 2023-08-01 18:10:12,828 [  INFO  ] Registering for moving messages - __main__
saic-mqtt-gateway  | 2023-08-01 18:10:13,129 [  INFO  ] Registering for region messages - __main__
saic-mqtt-gateway  | 2023-08-01 18:10:13,433 [  INFO  ] Registering for engineStart messages - __main__
saic-mqtt-gateway  | 2023-08-01 18:10:13,723 [  INFO  ] Registering for startVehicleStatus messages - __main__
saic-mqtt-gateway  | 2023-08-01 18:10:14,038 [  INFO  ] Registering for offCar messages - __main__
saic-mqtt-gateway  | 2023-08-01 18:10:14,325 [  INFO  ] Registering for speeding messages - __main__
saic-mqtt-gateway  | 2023-08-01 18:10:14,423 [  INFO  ] Added job "Check for new messages" to job store "default" - apscheduler.scheduler
saic-mqtt-gateway  | 2023-08-01 18:10:24,456 [  INFO  ] Setting active query interval in vehicle handler for VIN #VIN# to 0:00:30 - vehicle
saic-mqtt-gateway  | 2023-08-01 18:10:24,457 [  INFO  ] Setting charging query interval in vehicle handler for VIN #VIN# to 0:00:00 - vehicle
saic-mqtt-gateway  | 2023-08-01 18:10:24,458 [  INFO  ] Setting after shutdown query interval in vehicle handler for VIN #VIN# to 0:02:00 - vehicle
saic-mqtt-gateway  | 2023-08-01 18:10:24,462 [  INFO  ] Setting inactive query interval in vehicle handler for VIN #VIN# to 1 day, 0:00:00 - vehicle
saic-mqtt-gateway  | 2023-08-01 18:10:24,463 [  INFO  ] Setting charging query interval in vehicle handler for VIN #VIN# to 0:00:00 - vehicle
saic-mqtt-gateway  | 2023-08-01 18:10:24,464 [  INFO  ] Setting refresh mode to periodic - vehicle
saic-mqtt-gateway  | 2023-08-01 18:10:24,466 [  INFO  ] Updating vehicle status - __main__
saic-mqtt-gateway  | 2023-08-01 18:10:40,646 [  INFO  ] Updating charging status - __main__
saic-mqtt-gateway  | 2023-08-01 18:10:56,803 [  INFO  ] Setting charging query interval in vehicle handler for VIN #VIN# to 0:16:00 - vehicle
saic-mqtt-gateway  | 2023-08-01 18:10:56,803 [  INFO  ] Refreshing vehicle status succeeded... - __main__
saic-mqtt-gateway  | 2023-08-01 18:11:14,425 [  INFO  ] Running job "Check for new messages (trigger: interval[0:01:00], next run at: 2023-08-01 18:12:14 UTC)" (scheduled at 2023-08-01 18:11:14.423227+00:00) - apscheduler.executors.default
saic-mqtt-gateway  | 2023-08-01 18:11:14,704 [  INFO  ] 0 messages received - __main__
saic-mqtt-gateway  | 2023-08-01 18:11:14,704 [  INFO  ] Job "Check for new messages (trigger: interval[0:01:00], next run at: 2023-08-01 18:12:14 UTC)" executed successfully - apscheduler.executors.default
```

Thanks! Closing as completed.
